vochain/scrutinizer/vote.go

```diff
@@ -255,25 +255,30 @@ func (s *Scrutinizer) ComputeResult(processID []byte) error {
 		return fmt.Errorf("computeResult: cannot load processID %x from database: %w", processID, err)
 	}
 
-	// Compute the results
-	var results *indexertypes.Results
-	if results, err = s.computeFinalResults(p); err != nil {
-		return err
-	}
-	p.HaveResults = true
-	p.FinalResults = true
-	if err := s.db.Update(processID, p); err != nil {
-		return fmt.Errorf("computeResult: cannot update processID %x: %w, ", processID, err)
-	}
-	s.addVoteLock.Lock()
-	defer s.addVoteLock.Unlock()
-	if err := s.db.Upsert(processID, results); err != nil {
-		return err
-	}
+	if err := s.db.UpdateMatching(&indexertypes.Process{},
+		badgerhold.Where("ID").Eq(processID),
+		func(record interface{}) error {
```

Comment:
The computeFinalResults operation is expensive and the one we don't want to have inside the atomic operation, so the code would be something like this:

```
diff --git a/vochain/scrutinizer/vote.go b/vochain/scrutinizer/vote.go
index 6ab2f373..4a39a5cc 100644
--- a/vochain/scrutinizer/vote.go
+++ b/vochain/scrutinizer/vote.go
@@ -260,9 +260,17 @@ func (s *Scrutinizer) ComputeResult(processID []byte) error {
        if results, err = s.computeFinalResults(p); err != nil {
                return err
        }
-       p.HaveResults = true
-       p.FinalResults = true
-       if err := s.db.Update(processID, p); err != nil {
+       if err := s.db.UpdateMatching(&indexertypes.Process{},
+               badgerhold.Where("ID").Eq(processID),
+               func(record interface{}) error {
+                       update, ok := record.(*indexertypes.Process)
+                       if !ok {
+                               return fmt.Errorf("record isn't the correct type! Wanted Result, got %T", record)
+                       }
+                       update.HaveResults = true
+                       update.FinalResults = true
+                       return nil
+               }); err != nil {
                return fmt.Errorf("computeResult: cannot update processID %x: %w, ", processID, err)
        }
        s.addVoteLock.Lock()
```